### PR TITLE
NOISSUE - Fix ViewGroup and UpdateGroup

### PR DESCRIPTION
**Pull request by manuio**
Signed-off-by: Manuel Imperiale <manuel.imperiale@gmail.com>

### CODE REVIEW

File: users/api/requests.go [hunk 1]
```diff
@@ -129,6 +129,7 @@ func (req createGroupReq) validate() error {
 
 type updateGroupReq struct {
 	token       string
+	id          string
 	Name        string                 `json:"name,omitempty"`
 	Description string                 `json:"description,omitempty"`
```
Comment (mteodor) [inline users/api/requests.go:135]: we need parent id here also

File: users/api/requests.go [hunk 2]
```diff
@@ -138,12 +140,16 @@ func (req updateGroupReq) validate() error {
...
 	}
+	if req.id == "" {
+		return users.ErrMalformedEntity
+	}
 	if req.Name == "" {
 		return users.ErrMalformedEntity
 	}
 	if len(req.Name) > maxNameSize {
 		return users.ErrMalformedEntity
 	}
```
Comment (dborovcanin) [inline users/api/requests.go]: You can `or` these conditions.

File: users/api/endpoint.go [hunk 3]
```diff
@@ -259,23 +259,20 @@ func updateGroupEndpoint(svc users.Service) endpoint.Endpoint {
 	return func(ctx context.Context, request interface{}) (interface{}, error) {
 		req := request.(updateGroupReq)
 		if err := req.validate(); err != nil {
-			return createGroupRes{}, err
+			return updateGroupRes{}, err
 		}
+
 		group := users.Group{
 			Name:        req.Name,
```
Comment (mteodor) [inline users/api/endpoint.go:267]: mising ID

**Review by mteodor — DISMISSED** [review]

**Review by drasko — COMMENTED** [review]
OK for me, just take into account @dusanb94 's remark and I'll approve

**Review by drasko — APPROVED** [review]
LGTM

File: users/api/requests.go (hunk 2 above)
Comment (dborovcanin) [reply users/api/requests.go]: I meant that you can `or` all of these checks: `if id == "" || name == "" || len(name) > maxNameLen`.

Comment (manuio) [reply users/api/requests.go]: I agreed to group this checks by type but for consistency with other services and readability I would keep it like that.
You can check here: https://github.com/mainflux/mainflux/blob/master/things/api/things/http/requests.go

**Review by dborovcanin — APPROVED** [review]

**Pull request merged**